csrc/scheduler/normalization_utils.cpp

```diff
@@ -659,50 +661,51 @@ std::optional<int64_t> getOptionalInnerOuterPersistentBufferBatches(
   //! in the reduction domain. vectorization_factor is the vectorization factor
   //! of inputs and outputs.
   auto getMaximumInnerOuterPersistentBufferBatch = [&]() -> int64_t {
-    int64_t register_per_batch = persistent_buffer_size / inner_dim_numel /
-        scheduler_utils::bytes_per_register * vectorize_factor;
+    int64_t register_per_batch = persistent_buffer_size / inner_dim_numel *
+        vectorize_factor / scheduler_utils::bytes_per_register;
     return scheduler_utils::safeDiv(
         scheduler_utils::max_registers_per_thread -
             scheduler_utils::register_overhead,
         register_per_batch);
   };
 
-  int64_t threads_per_block = warp_size / 4;
-  int64_t inner_batch = inner_dim_numel / vectorize_factor / threads_per_block;
-  const int64_t threads_per_block_max = inner_dim_numel >= 20480l ? 512l : 256l;
+  const int64_t after_vectorization = inner_dim_numel / vectorize_factor;
+  const int64_t threads_per_block_min = std::min(after_vectorization, 128l);
+  const int64_t threads_per_block_max = getThreadsPerSMGivenRegPerThread(255l);
   const int64_t batch_min = getMinimumBatch();
-  auto canReduceBatch = [&](auto factor) -> bool {
-    return inner_batch % factor == 0 && inner_batch / factor >= batch_min &&
-        threads_per_block * factor <= threads_per_block_max;
-  };
-  // reduce batch size by 2, 3, or 5 until we reach the minimum batch size
-  // only consider divisible splits
-  while (inner_batch > batch_min && threads_per_block < threads_per_block_max) {
-    bool modified = false;
-    for (auto factor : {2, 3, 5}) {
-      if (canReduceBatch(factor)) {
-        inner_batch /= factor;
-        threads_per_block *= factor;
-        modified = true;
-        break;
-      }
-    }
-    if (!modified) {
-      break;
-    }
-  }
-  TORCH_INTERNAL_ASSERT(
-      inner_batch >= batch_min,
-      "Batch size is larger than expected! Please report this issue.",
-      " inner_batch = ",
-      inner_batch,
-      ", batch_min = ",
-      batch_min);
   const int64_t batch_max = getMaximumInnerOuterPersistentBufferBatch();
-  if (inner_batch <= batch_max) {
-    return inner_batch;
+
+  // Start from the smallest block size. If the corresponding batch size is
```

Comment:
nit: "block size" and "batch size" can be confusing. Maybe "thread block size" or "CTA size"